src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -354,5 +451,53 @@ private static VisualStudioInstance GetDevConsoleInstance()
             return null;
         }
 #endif
+
+        // Taken from MSBuild/NativeMethodsShared
+        private static readonly object IsRunningOnMonoLock = new object();
+
+        private static bool? _isRunningOnMono;
+
+        /// <summary>
+        /// Gets a flag indicating if we are running under MONO
+        /// </summary>
+        internal static bool IsRunningOnMono
+        {
+            get
+            {
+                if (_isRunningOnMono != null) return _isRunningOnMono.Value;
+
+                lock (IsRunningOnMonoLock)
+                {
+                    if (_isRunningOnMono == null)
+                    {
+                        // There could be potentially expensive TypeResolve events, so cache IsMono.
+                        // Also, VS does not host Mono runtimes, so turn IsMono off when msbuild is running under VS
```

Comment:
I assume you're just talking about VS on Windows right?

Reply:
yep